Samples.Objc/Samples.ObjC/API/Requests/CreateOrderRequest.m

```diff
@@ -1,44 +1,107 @@
 //
 //  CreateOrderRequest.m
-//  PayPalCheckout-Samples-iOS-Objc
+//  PayPalNativeCheckoutObjC
 //
-//  Created by Haider Khan on 5/18/20.
-//  Copyright © 2020 PayPal. All rights reserved.
+//  Created by Nguyen, The Nhat Minh on 3/14/21.
 //
 
 #import "CreateOrderRequest.h"
 
 @implementation CreateOrderRequest
 
-- (id)initWith:(OrderIntent)intent purchaseUnits:(NSArray *)purchaseUnits {
+- (id)initWithOrder:(PPCOrderRequest*)order andAccessToken:(NSString*)token {
   self = [super init];
   if (self) {
-    self.intent = intent;
-    self.purchaseUnits = purchaseUnits;
+    self.order = order;
+    self.accessToken = token;
   }
   return self;
 }
 
 - (NSDictionary *)properties {
+  NSString *intentString = @"";
+  if (intentString == PPCOrderIntentCapture) {
+    intentString = @"CAPTURE";
+  } else {
+    intentString = @"AUTHORIZE";
+  }
```

Comment:
This will not work as intended because `PPCOrderIntentCapture` is a enum of type `Int`. Also `==` in objective-c does an equality check on pointers and not the contents of the pointer. The correct use for strings would be `[intentString isEqualToString:@"my other string"]`.

What we most likely want is this:

```objectivec
NSString *intentString = @"";
if (_order.intent == PPCOrderIntentCapture) {
  intentString = @"CAPTURE";
} else if (_order.intent == PPCOrderIntentAuthorize) {
  intentString = @"AUTHORIZE";
}
```

Since `_order.intent` is of type enum which extends `Int` we can do `==` since it works on raw primitive types ok.